students/JimJenkins/session04/file_exercise.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+"""
+Author: Jim Jenkins (dvlupr)
+Date: 11/05/2018
+
+    Write a program which prints the full path for all files in the current directory, one per line
+    Write a program which copies a file from a source, to a destination (without using shutil, or the OS copy command).
+        Advanced: make it work for any size file: i.e. don’t read the entire contents of the file into memory at once.
+        This should work for any kind of file, so you need to open the files in binary mode: open(filename, 'rb')
+        (or 'wb' for writing). Note that for binary files, you can’t use readline() – lines don’t have any meaning for
+        binary files.
+        Test it with both text and binary files (maybe jpeg or something of your choosing).
+
+"""
+
+import os
+
+#list full path name for files in the cwd
+cwd = os.getcwd()
+files = os.listdir(cwd)
+
+for file in files:
+    print(cwd + '\\' + file)
+
+
+#copy file
+ori_file = 'students.txt'
+src_file = open(ori_file, 'r')
```

Comment:
open opens in text mode be default -- so this will nly work on text files -- it may barf if you try to use this with a  binary file.

try a  JPEG or something as an example.